drivers/SmartThings/matter-lock/src/init.lua

```diff
@@ -519,6 +521,15 @@ end
 local function device_init(driver, device)
   device:set_component_to_endpoint_fn(component_to_endpoint)
   device:subscribe()
+
+  -- Device may require pin for remote operation if it supports COTA and PIN features.
+  local eps = device:get_endpoints(DoorLock.ID, {feature_bitmap = DoorLock.types.DoorLockFeature.CREDENTIALSOTA | DoorLock.types.DoorLockFeature.PIN_CREDENTIALS})
+  if #eps == 0 then
+    device.log.debug("Device will not require PIN for remote operation")
+    device:set_field(lock_utils.COTA_CRED, false, {persist = true})
+  else
+    device:send(DoorLock.attributes.RequirePINforRemoteOperation:read(device, eps[1]))
+  end
```

Comment:
Can we get format this repeat code into a local function and call it in `device_init` and `device_added`?

Reply:
The reason I kept it separate was so that I could merge the read with the other reads in `added` and then I could just send the single request in `init`. I can try to find a way to neatly return a read request if needed, but either way there will be a check to merge it with the existing read req in `added` or just send it as a single read in `init`

Reply:
Or I guess we could just send two requests